src/Pho/Server/Rest/Router.php

```diff
@@ -11,88 +11,111 @@
+            // this lambda converts some common usages such as uuid
+            // for unique cases, regex must be kept at the 
+            // route level.
+            return str_replace("{uuid}", "{uuid:[0-9a-fA-F]{32}}", $path);
+        };
+        $this->dispatcher = \FastRoute\simpleDispatcher(function(\FastRoute\RouteCollector $r) use ($routes, $resolvePath) {
+
+            // load kernel routes at last
+            uksort($routes, function ($a, $b) {
+                return $a === 'kernel';
+            });
+
+            foreach($routes as $controller => $handlers) {
+                foreach($handlers as $handler => $route) {
+                    $method = $route[0];
+                    $path = $resolvePath($route[1]);
+                    $r->addRoute($method, $path, "{$controller}::{$handler}");
+                }
+            }
+        });
+        return $this;
     }
 
-    public static function initDelete(Server $server, array $controllers): void
+    public function __invoke(ServerRequestInterface $request)
     {
-        $server->delete('{uuid}', [$controllers["entity"], "delete"])
-            ->where('uuid', '[0-9a-fA-F]{32}');
-        $server->delete('{uuid}/attribute/{key}', [$controllers["entity"], "deleteAttribute"])
-            ->where('uuid', '[0-9a-fA-F]{32}');
+        $method = $request->getMethod(); // GET or POST
+        $path = $request->getUri()->getPath(); // the path
+        $routeInfo = $this->dispatcher->dispatch($method, $path);   
+        switch ($routeInfo[0]) {
+            
+            case Dispatcher::NOT_FOUND:
+                $response = $this->controllers["kernel"]->fail("Not Found", 404);
+                break;
+            case Dispatcher::METHOD_NOT_ALLOWED:
+                $allowedMethods = $routeInfo[1];
+                $response = $this->controllers["kernel"]->fail("Allowed Methods: ".join(', ', array_unique($allowedMethods)), 405);
+                break;
+                
+            case Dispatcher::FOUND:
+                $handler = $routeInfo[1];
+                $vars = $routeInfo[2];
+                $this->kernel->logger()->info("Found: ".$handler . " --- ". print_r($vars, true));
+                // ... call $handler with $vars
+                // @todo This must be implemented
+                list($controllerName, $methodName) = explode('::', $routeInfo[1]);
+                if (null === $controller = $this->controllers[$controllerName] ?? null) {
+                    throw new \RuntimeException("Controller $controllerName not found");
+                }
+                if (! method_exists($controller, $methodName)) {
+                    throw new \RuntimeException("Method $methodName does not exist in controller $controllerName");
+                }
+                $response = new Response(200, AbstractController::HEADERS, null);
+                $response = call_user_func_array([ $controller, $methodName ], array_merge([ $request, $response ], $vars));
```

Comment:
I don't see how you deal with json vs jsonp case here. See AbstractController.php.

Reply:
cc/ @codekaar

Reply:
json vs jsonp case is not handled here. Instead, it is handled in `AbstractController::respond`